### PR TITLE
WIP: support teleport as kubeconfig client

### PR DESCRIPTION
**Pull request by anvddriesch**
Towards https://github.com/giantswarm/giantswarm/issues/30767

Instead of using the secret stored on the MC we want to enable kube access via teleport

### CODE REVIEW

File: framework.go [hunk 1]
```diff
@@ -240,12 +252,22 @@ func (f *Framework) ApplyBuiltCluster(ctx context.Context, builtCluster *applica
...
+	err := wait.For(wait.IsClusterReadyCondition(ctx, c, clusterName, namespace), wait.WithContext(ctx), wait.WithInterval(10*time.Second))
 	if err != nil {
 		return nil, err
 	}
 
-	return client.NewFromSecret(ctx, f.MC(), clusterName, namespace)
+	return client.NewFromClient(ctx, c, clusterName, namespace)
+}
+
+func (f *Framework) getKubeConfigClient() any {
```
Comment (AverageMarcus) [inline framework.go:265]: Please can we not use `any` type. If needed we sure use an interface.

File: pkg/env/const.go [hunk 2]
```diff
@@ -5,6 +5,8 @@ const (
 	// will be used to connect to the MC
 	Kubeconfig = "E2E_KUBECONFIG"
 
+	TeleportIdentityFile = "TELEPORT_IDENTITY_FILE"
```
Comment (AverageMarcus) [inline pkg/env/const.go:8]: Doc comment please :)

File: pkg/client/client.go [hunk 3]
```diff
@@ -74,6 +75,17 @@ func NewFromRawKubeconfig(kubeconfig string) (*Client, error) {
 	return newClient(restConfig, clusterName)
 }
 
+func NewFromClient(ctx context.Context, client any, clusterName string, namespace string) (*Client, error) {
```
Comment (AverageMarcus) [inline pkg/client/client.go:78]: Doc comment please :)

File: framework.go (hunk 1 above)
Comment (anvddriesch) [reply framework.go:265]: Don't worry, this is stil WIP :) I will be adding an interface for the kubeconfig client

Comment (AverageMarcus) [reply framework.go:265]: Ah! Sorry. I saw the request for review and dove in 😉

Comment (AverageMarcus) [reply framework.go:265]: Wanna ping me when it's good to go and I'll take a look?

Comment (anvddriesch) [reply framework.go:265]: Yeah definitely! Sorry for the premature ping, I'm used to the draft button being the default :sweat_smile:

Comment (AverageMarcus) [reply framework.go:265]: No worries :) I like to keep on top of notifications when they come in. I didn't spot the WIP in the title.

**Comment by anvddriesch** [comment]
Closing as teleport api is not able to give us what we want. We will use tbot https://github.com/giantswarm/clustertest/pull/244

**Pull request closed without merging**